server/src/resolvers/query.js

```diff
@@ -1,140 +1,34 @@
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 2,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'What is meaning of life?',
-        description: 'Sometimes we thinking about huge size of the universe. ' +
-            'Compare to what humanity accomplished so far,' +
-            ' the universe seems wast of space... isn\' it?',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 3,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'If AI gets self awareness...',
-        description: 'That means bad thing to human? YES.' +
-            ' But in a perspective of evolution? NO.',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 4,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'When Space X going to launch a rocket to the Mars?',
-        description: 'Let\'s analyse the probability!',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 5,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'title1',
-        description: 'description1',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 6,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'title2',
-        description: 'description2',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 7,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'title3',
-        description: 'description3',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-      {
-        id: 8,
-        host: 0,
-        creationTime: '',
-        lastUpdatedTime: '',
-        schedule: [],
-        title: 'title4',
-        description: 'description4',
-        price: 999999.99,
-        maxParticipants: 1000,
-        tags: [],
-        participants: [],
-      },
-    ];
-
-    let ret = [];
-    let max_size = Math.min(events_list.length, pageSize);
-    max_size = Math.min(max_size, 9);
-    ret = events_list.slice(0, max_size);
-
+  getTagNames: async (_, {after, pageSize}, {dataSources, userId})=>{
+    const tagNames = await dataSources.mainAPI.findTagName(after, pageSize);
     return {
-      cursor: 8,
-      events: ret,
+      cursor: after + pageSize,
+      tagNames: tagNames,
```

Comment:
Let's read this(https://medium.com/paypal-engineering/graphql-resolvers-best-practices-cd36fdbcef55)